src/txmempool.cpp

```diff
@@ -721,7 +744,7 @@ void CTxMemPool::removeForBlock(const std::vector<CTransaction>& vtx, unsigned i
 
```

Comment:
In removeForBlock we must remove the BTRs (and all his deps) of a sidechain whose safeguard window finishes at this block.